scripts/environment/gsutil_env.py

```diff
@@ -0,0 +1,281 @@
+                            'Continue anyway? y/N ').strip().lower()
+                        if same_bucket != 'y':
+                            continue
+                    self.output_bucket = bucket
+
+                print 'Using bucket: {}\n'.format(bucket)
+                break
+
+    def _create_or_reuse_bucket(self, object_type):
+        """
+        Give the user the option to either create or re-use a bucket.
+
+        This method can be used for either the POLICY or the OUTPUT
+        bucket types.
+        """
+        bucket_name = None
+
+        default_bucket = ''
+        if object_type == ScannerBucketObject.POLICY:
+            default_bucket = self.DEFAULT_BUCKETNAME_POLICIES
+        elif object_type == ScannerBucketObject.OUTPUT:
+            default_bucket = self.DEFAULT_BUCKETNAME_OUTPUT
+
+        default_bucket = default_bucket.format(self.gcloud_config.project_id)
+
+        while True:
+          bucket_name = raw_input('Enter a bucket name (default: {}): '
+                                  .format(default_bucket)).strip()
+          if not bucket_name:
+              bucket_name = default_bucket
+
+          if not bucket_name.startswith('gs://'):
+              bucket_name = 'gs://{}'.format(bucket_name)
+
+          p = Popen(['gsutil', 'mb', bucket_name], stdout=PIPE, stderr=PIPE)
+          out, err = p.communicate()
+          if p.returncode:
+              print err
+              if self._force_use_bucket(bucket_name, err):
+                  break
+
+        return bucket_name
+
+    def _force_use_bucket(self, bucket_name, mb_error_text):
+        """
+        Parse the error text from gsutil mb and check what the
+        exception is. If it's a 409, confirm with the user whether
+        they actually want to use that bucket name.
+        """
+        exceptions = self.GCS_LS_ERROR_REGEX.findall(mb_error_text)
+        if len(exceptions):
+          try:
+              mb_exception, res_code_mb = exceptions[0]
+              if mb_exception == 'ServiceException' and int(res_code_mb) == 409:
+                  p = Popen(['gsutil', 'ls', bucket_name],
+                            stdout=PIPE, stderr=PIPE)
+                  ls_out, ls_err = p.communicate()
+                  ls_exceptions = self.GCS_LS_ERROR_REGEX.findall(ls_err)
+                  if p.returncode:
+                      print ls_err
+
+                  if len(ls_exceptions):
+                      ls_exception, res_code_ls = ls_exceptions[0]
+                      if int(res_code_ls) == 403:
+                          print 'You can\'t use this bucket.'
+                          return False
+
+              while True:
+                should_use = raw_input(
+                    'Re-use the bucket "{}"? (y/N) '
+                    .format(bucket_name)).strip().lower()
+
+                if should_use == 'y':
+                    return True
+                elif should_use == 'n':
+                    break
+          except:
```

Comment:
exception handling is usually preferred to be targeted, rather than a blanket catch all